android/main/java/com/couchbase/lite/internal/replicator/CBLWebSocket.java

```diff
@@ -47,9 +47,10 @@
         int port,
         String path,
         Map<String, Object> options,
+        Fn.Provider<CBLCookieStore> cookieStoreProvider,
```

Comment:
I can't really tell.  Do you need a provider here or could you just pass the CookieStore?  Is there a reason you have to create a new on for every request?

Reply:
I think there is no need for the Provider. I don't know why now I'm using that. Probably because I like your Fn.Consumer.